pennylane/devices/default_mixed.py

```diff
@@ -52,15 +52,94 @@
 from ._qubit_device import QubitDevice
 
 # We deliberately separate the imports to avoid confusion with the legacy device
+import warnings
+from collections.abc import Callable, Sequence
+from dataclasses import replace
 from typing import Optional
+
+from pennylane.ops.channel import __qubit_channels__ as channels
+from pennylane.transforms.core import TransformProgram
 from pennylane.tape import QuantumScript
+
 from . import Device
-from .execution_config import ExecutionConfig
+from .execution_config import DefaultExecutionConfig, ExecutionConfig
+from .preprocess import (
+    decompose,
+    no_sampling,
+    null_postprocessing,
+    validate_device_wires,
+    validate_measurements,
+    validate_observables,
+)
 from .modifiers import simulator_tracking, single_tape_support
 
 logger = logging.getLogger(__name__)
 logger.addHandler(logging.NullHandler())
 
+observables = {
+    "Hadamard",
+    "Hermitian",
+    "Identity",
+    "PauliX",
+    "PauliY",
+    "PauliZ",
+    "Prod",
+    "Projector",
+    "SProd",
+    "Sum",
+}
+
+
+def observable_stopping_condition(obs: qml.operation.Operator) -> bool:
+    """Specifies whether an observable is accepted by DefaultQubitMixed."""
+    if obs.name in {"Prod", "Sum"}:
+        return all(observable_stopping_condition(observable) for observable in obs.operands)
+    if obs.name == "LinearCombination":
+        return all(observable_stopping_condition(observable) for observable in obs.terms()[1])
+    if obs.name == "SProd":
+        return observable_stopping_condition(obs.base)
+
+    return obs.name in observables
+
+
+def stopping_condition(op: qml.operation.Operator) -> bool:
+    """Specify whether an Operator object is supported by the device."""
+    expected_set = DefaultMixed.operations | {"Snapshot"} | channels
+    return op.name in expected_set
+
+
+def stopping_condition_shots(op: qml.operation.Operator) -> bool:
+    """Specify whether an Operator object is supported by the device with shots."""
+    return stopping_condition(op)
```

Comment:
We can simply just not provide a `stopping_condition_shots`.  For `DefaultQubit`, it was kinda a hack that might not be necessary anymore.